packages/shared/src/write-archive/index.ts

```diff
@@ -36,7 +36,13 @@ export async function writeTestResult(
   archive: ResourceArchive,
   chromaticStorybookParams: ChromaticStorybookParameters
 ) {
-  const { title, outputDir, pageUrl } = e2eTestInfo;
+  const { titlePath, outputDir, pageUrl } = e2eTestInfo;
+  // remove the test file extensions (.spec.ts|ts, .cy.ts|js), preserving other periods in directory, file name, or test titles
+  const titlePathWithoutFileExtensions = titlePath.map((pathPart) =>
```

Comment:
There's no guarantee that these files contain `.spec.` or `.cy.`. We should ensure that we still parse off the final `.ts/.js/etc` extension if they don't.

Reply:
For playwright: https://playwright.dev/docs/test-configuration#filtering-tests

Reply:
We probably also want to expand the extensions to `ts|js|mjs|cjs`. Anything else?

Reply:
Fair point, I'll add those in. I bet [those extensions](https://github.com/chromaui/chromatic-e2e/pull/98#discussion_r1480356480) you mentioned are good enough for now, and if someone uses something unusual we can add it later!